### PR TITLE
Provide synonims for some data types

### PR DESCRIPTION
**Pull request by Potya**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Provide synonims for some data types

Detailed description / Documentation draft:

Provide NUMERIC as a synonim to DECIMAL; provide many other similar symonims. 
"LONG", "String"
"NUMERIC", "Decimal"
"FIXED", "Decimal"
"BOOL", "Int8"
"BOOLEAN", "Int8"
"INT1", "Int8"
"INT2", "Int16"
"INT4", "Int32"
"INT8", "Int64"

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Ok, but we have to remove at least INT8 because it will be ambiguous with Int8.

And also please check whether these types
```
"INT1", "Int8"
"INT2", "Int16"
"INT4", "Int32"
```
are standard or not.

File: src/Interpreters/InterpreterSelectQuery.cpp [hunk 1]
```diff
@@ -2280,6 +2280,8 @@ void InterpreterSelectQuery::executeUnion(Pipeline & pipeline, Block header)
         pipeline.streams.push_back(pipeline.stream_with_non_joined_data);
         pipeline.stream_with_non_joined_data = nullptr;
     }
+
```
Comment (alexey-milovidov) [inline src/Interpreters/InterpreterSelectQuery.cpp:2283]: Need to remove unrelated changes.

File: src/DataTypes/DataTypeString.cpp [hunk 2]
```diff
@@ -380,6 +380,7 @@ void registerDataTypeString(DataTypeFactory & factory)
     factory.registerAlias("TEXT", "String", DataTypeFactory::CaseInsensitive);
     factory.registerAlias("TINYTEXT", "String", DataTypeFactory::CaseInsensitive);
     factory.registerAlias("MEDIUMTEXT", "String", DataTypeFactory::CaseInsensitive);
+    factory.registerAlias("LONG", "String", DataTypeFactory::CaseInsensitive);
```
Comment (alexey-milovidov) [inline src/DataTypes/DataTypeString.cpp:383]: Is `LONG` belongs to SQL standard or not?

**Comment by alexey-milovidov** [comment]
Also need to add `REAL`.
https://www.postgresql.org/docs/9.1/datatype-numeric.html

**Comment by alexey-milovidov** [comment]
Please provide some tests...

**Comment by amosbird** [comment]
I'm confused that `long` a string type

**Comment by alexey-milovidov** [comment]
Yes, it's string in MySQL and integer in MS Access. I have removed this synonim.
